migrations/versions/b9131d0686eb_.py

```diff
@@ -0,0 +1,85 @@
+"""Add a generic admin policy for tokenlist
+
+Revision ID: b9131d0686eb
+Revises: 849170064430
+Create Date: 2019-06-28 07:37:59.224088
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'b9131d0686eb'
+down_revision = '849170064430'
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import orm
+from sqlalchemy.schema import Sequence
+from privacyidea.lib.policy import SCOPE, ACTION
+
+Base = declarative_base()
+
+# dashes are not allowed, when creating policies in the WebUI
+# or via the library. So we are sure, that in normal operation
+# this policy can never be created.
+POLICYNAME = u"pi-update-policy-b9131d0686eb"
```

Comment:
This idea to avoid name clashes is clever :) But I think we should avoid using dashes, because it contradicts our assumption that policy names never contain dashes, and this may cause trouble in the future.
As a current side effect, we cannot edit this policy via the web UI.
I think something like "policy_migration_b9131d0686eb" should be OK too.

Reply:
I did not plan on editing the policy.
But I would think of deleting the policy.
This is a nice side effect, that the policy can only be deleted. Do you not like this?

Reply:
I guess the side effect is OK, but I fear that there may be some point in the future at which we do something relying on the assumption that all policy names only contain letters, digits, underscores, ... and then run into trouble because there are some installations with a policy not matching this rule. :-/

Reply:
...and are you sure, that no customer until now has created a policy named "policy_migration_b9131d0686eb" because this would realy be problematic since we overwrite an existing policy.

Reply:
Not 100% of course :) But maybe we should only create the migration policy in case a policy with that name doesn't exist yet anyway?

Reply:
If you do not create the policy, then migration will fail, since the policy-logic is missing.
With the current setup we will only create the policy, if the migration has not yet occurred, i.e. the data base schema is on an older version.

Reply:
Hm well, I think *if* someone has configured a policy named ``policy_migration_b9131d0686eb`` and we do not create our policy during migration, the migration itself will work, but the admins would have a restricted token list view afterwards?
I think if this unlikely case happens, it would be OK to require manually adding another migration policy.

Reply:
Should we just go with the way to allow "-" in the policy name as discussed in person?

Reply:
As it turns out, dashes in policy names are not really a problem for privacyIDEA. We are not allowed to modify them, but we can enable/disable/delete them.

As it's nice to be able to create policies in migration scripts that don't clash with user-defined policies, let's keep the dashes in the policy name.

Do we have a good place to document this behavior? e.g. with a sentence
"Dashes are normally not allowed in policy names. However, privacyIDEA might create policies whose names contain dashes during upgrades".

Reply:
I think we should document this at the update section and at the policies.

Reply:
Look, it is bbc24af5

Reply:
Thanks!